modules/core/src/test/scala/org/tessellation/domain/statechannel/StateChannelValidatorSuite.scala

```diff
@@ -0,0 +1,100 @@
+package org.tessellation.domain.statechannel
+
+import cats.data.Validated.Valid
+import cats.effect.IO
+import cats.effect.kernel.Resource
+import cats.syntax.validated._
+
+import org.tessellation.dag.dagSharedKryoRegistrar
+import org.tessellation.dag.snapshot.StateChannelSnapshotBinary
+import org.tessellation.domain.aci.StateChannelOutput
+import org.tessellation.domain.statechannel.StateChannelValidator
+import org.tessellation.domain.statechannel.StateChannelValidator.{InvalidSigned, NotSignedByAddressOwner}
+import org.tessellation.ext.kryo._
+import org.tessellation.keytool.KeyPairGenerator
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.key.ops.PublicKeyOps
+import org.tessellation.security.signature.Signed.forAsyncKryo
+import org.tessellation.security.signature.SignedValidator
+import org.tessellation.security.signature.SignedValidator.InvalidSignatures
+import org.tessellation.shared.sharedKryoRegistrar
+
+import eu.timepit.refined.auto._
+import org.scalacheck.Arbitrary
+import weaver.MutableIOSuite
+
+object StateChannelValidatorSuite extends MutableIOSuite {
+
+  type Res = (KryoSerializer[IO], SecurityProvider[IO])
+
+  override def sharedResource: Resource[IO, StateChanneServiceSuite.Res] =
+    KryoSerializer.forAsync[IO](dagSharedKryoRegistrar.union(sharedKryoRegistrar)).flatMap { ks =>
+      SecurityProvider.forAsync[IO].map((ks, _))
+    }
+
+  private val stateChannel = Arbitrary.arbitrary[StateChannelSnapshotBinary].sample.get
+
+  test("should succeed when state channel is signed correctly") { res =>
+    implicit val (kryo, sp) = res
+
+    val validator = mkValidator()
+
+    for {
+      keyPair <- KeyPairGenerator.makeKeyPair[IO]
+      signedSCBinary <- forAsyncKryo(stateChannel, keyPair)
+      scOutput = StateChannelOutput(keyPair.getPublic().toAddress, signedSCBinary)
+      result <- validator.validate(scOutput)
+    } yield expect.same(Valid(scOutput), result)
+
+  }
+
+  test("should fail when the signature is wrong") { res =>
+    implicit val (kryo, sp) = res
+
+    val validator = mkValidator()
+
+    for {
+      keyPair1 <- KeyPairGenerator.makeKeyPair[IO]
+      keyPair2 <- KeyPairGenerator.makeKeyPair[IO]
+      signedSCBinary <- forAsyncKryo(stateChannel, keyPair1).map(signed =>
+        signed.copy(proofs = signed.proofs.map(_.copy(id = keyPair2.getPublic.toId)))
+      )
+      scOutput = StateChannelOutput(keyPair2.getPublic().toAddress, signedSCBinary)
+      result <- validator.validate(scOutput)
+    } yield
+      expect.same(
+        InvalidSigned(InvalidSignatures(signedSCBinary.proofs)).invalidNec,
+        result
+      )
+  }
+
+  test("should fail when the address is wrong") { res =>
```

Comment:
when the address is wrong or when the signature is wrong, because of signature<->address mismatch?
in the latter case, imo address is anyway correct

Reply:
Changed.